File: BaekjoonAlgorithm/BA_9251/BJAlgo_9251.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class BJAlgo_9251 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String stringA = br.readLine();
        String stringB = br.readLine();
        int[][] result = new int[stringB.length() + 1][stringA.length() + 1];

        for (int i =  1; i < result.length; i++) {
            for (int j = 1; j < result[i].length; j++) {
                if (stringB.charAt(i - 1) != stringA.charAt(j - 1)) {
                    result[i][j] = Math.max(result[i - 1][j], result[i][j - 1]);
                } else {
                    result[i][j] = result[i - 1][j - 1] + 1;
                }
            }
        }

        System.out.println(result[stringB.length()][stringA.length()]);
    }
}

File: BaekjoonAlgorithm/BA_9251/BJAlgo_9251.py
from sys import stdin

stringA = stdin.readline().strip()
stringB = stdin.readline().strip()

result = [[0] * (len(stringA) + 1) for _ in range(len(stringB) + 1)]
for i in range(1, len(result)):
    for j in range(1, len(result[i])):
        if stringB[i - 1] != stringA[j - 1]:
            result[i][j] = max(result[i - 1][j], result[i][j - 1])
        else:
            result[i][j] = result[i - 1][j - 1] + 1

print(result[-1][-1])
